src/form.js

```diff
@@ -983,6 +983,21 @@ define([
             var _this = this;
             this._logicManager.forEachBrokenReference(updateReferences);
         },
+        isReferencedByOtherMugs: function (mug, except) {
```

Comment:
Just looked at your last commit. Looks like you pushed a bulk of this to the logic manager which is what I was thinking about :+1: